src/plugins/barista/tea.rs

```diff
@@ -0,0 +1,447 @@
+            ],
+            VesselType::Bowl => &[
+                "burl wood-and-silver tea",
+                "Tibetan tea",
+                "Tibetan silver tea",
+            ],
+            VesselType::Samovar => &["antique", "vintage", "brass", "silver"],
+            VesselType::Teacup => &[],
+            VesselType::Hohin => &[],
+            VesselType::Gaiwan => &["porcelain", "Ruyao"],
+            VesselType::Shiboridashi => &["porcelain", "red clay"],
+        }
+    }
+}
+
+impl fmt::Display for VesselType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            VesselType::Teapot => f.write_str("teapot"),
+            VesselType::Mug => f.write_str("mug"),
+            VesselType::Bowl => f.write_str("bowl"),
+            VesselType::Samovar => f.write_str("samovar"),
+            VesselType::Teacup => f.write_str("teacup"),
+            VesselType::Hohin => f.write_str("hohin"),
+            VesselType::Gaiwan => f.write_str("gaiwan"),
+            VesselType::Shiboridashi => f.write_str("shiboridashi"),
+        }
+    }
+}
+
+// tea adjectives. brands, places, etc.
+//
+// Kangra is a location in India where some kinds of green
+// tea are made.
+enum Adjective {
+    Newmans,
+    Earl,
+    FairTrade,
+    Homemade,
+    #[allow(dead_code)]
+    HomeBrewn,
+    Kangra,
+    Irish,
+    English,
+    Darjeel,
+    #[allow(dead_code)]
+    Vanilla,
+    #[allow(dead_code)]
+    Lemongrass,
+    #[allow(dead_code)]
+    Hibiscus,
+}
+
+impl fmt::Display for Adjective {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            Adjective::Newmans => f.write_str("Newman's Own"),
+            Adjective::Earl => f.write_str("Earl Grey"),
+            Adjective::FairTrade => f.write_str("fair trade"),
+            Adjective::Homemade => f.write_str("homemade"),
+            Adjective::HomeBrewn => f.write_str("home-brewn"),
+            Adjective::Kangra => f.write_str("Kangra"),
+            Adjective::Irish => f.write_str("Irish-breakfast"),
+            Adjective::English => f.write_str("English-breakfast"),
+            Adjective::Darjeel => f.write_str("Darjeeling"),
+            Adjective::Vanilla => f.write_str("vanilla"),
+            Adjective::Lemongrass => f.write_str("lemongrass"),
+            Adjective::Hibiscus => f.write_str("hibiscus"),
+        }
+    }
+}
+
+const SIZES: &[&str] = &[
+    "large", "small", "medium", "tall", "wide", "big", "100ml", "giant", "tiny",
+];
+const FILLED_WITH: &[&str] = &[
+    "filled with",
+    "of",
+    "stuffed with",
+    "full of",
+    "brimming with",
+];
+
+// Temperature of tea.
+const COLD: &[&str] = &["iced", "cold", "chilled", "ice cold"];
+const HOT: &[&str] = &[
+    "boiling",
+    "scalding",
+    "steaming",
+    "sweltering",
+    "toasty hot",
+];
+
+// It's unfortunate, but the best way to handle this (because there is no const
+// flatten) is manually copying them.
```

Comment:
Can you use [lazy_static](https://docs.rs/lazy_static/1.4.0/lazy_static/)? That'd clean this up a fair bit I think.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=05067fa0cc5f2cc0df29ccb0edb438f5

Reply:
Good call. This really cleaned it up.